daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/ExprParser.scala

```diff
@@ -18,16 +18,18 @@ private[parser] class ExprParser[P](parserParameters: ParserParameters[P]) {
   import typeParser._
 
   lazy val expr0: Parser[Expr] =
-    literal ^^ EPrimLit |
+    // expressions starting with fullIdentifier should come before litterals
+    eRecCon |
```

Comment:
Was the problem that we parsed it as a literal because that comes first here?

Reply:
Yes. 
PartyId and PackageId have the same syntax.

Reply:
thanks for the explanation!